internal/topology/check/compatibility_test.go

```diff
@@ -450,7 +450,7 @@ func TestClusterClassesAreCompatible(t *testing.T) {
 		},
 
 		{
-			name: "pass for compatible clusterClasses",
+			name: "pass for compatible clusterClasses from different NS",
```

Comment:
I think we should change the namespace of all templates in this ClusterClass (e.g. builder.InfrastructureClusterTemplate)

Reply:
Changed it everywhere to make it a standard scenario